File: easy/3.Rotate Array/go/main.go
package main

import "fmt"
import "slices"

func main() {
	x := []uint8{1, 2, 3, 4, 5, 6, 7}
	k := 3

	x = rotate(x, k)
	fmt.Printf("%v\n", x)
}

func rotate(slc []uint8, k int) []uint8 {
	k = (k + 1) % len(slc)
	var remainder = slc[:k]
	var rest = slc[k:]
	return slices.Concat(rest, remainder)
}

File: easy/3.Rotate Array/rust/src/main.rs
fn main() {
    println!("Hello, world!");
    let mut arr = vec![1, 2, 3, 4, 5, 6, 7];

    rotate(&mut arr, 3);
    dbg!(arr);
}

fn rotate(nums: &mut Vec<i32>, k: i32) {
    let r = k as usize % nums.len();

    let mid = nums.len() - r;

    let remainder = &nums[0..mid];
    let shifted = &nums[mid..nums.len()];
    let arr = [shifted, remainder].concat();
    *nums = arr.to_vec();
}
